### PR TITLE
Implement user migration

### PR DESCRIPTION
**Pull request by jonyoung123**

### CODE REVIEW

File: database/migrations/20221201070519-user-migrations.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      },
+      first_name: {
+        type: Sequelize.STRING,
+        allowNull: false,
+      },
+      last_name: {
+        type: Sequelize.STRING,
+        allowNull: false,
+      },
+      email_name: {
```
Comment (kelue) [inline database/migrations/20221201070519-user-migrations.js]: email should be email not email_name

File: database/migrations/20221201070519-user-migrations.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        type: Sequelize.STRING,
+        allowNull: false,
+        unique: true,
+      },
+      password: {
+        type: Sequelize.STRING,
```
Comment (kelue) [inline database/migrations/20221201070519-user-migrations.js:23]: password should have unique constraint too

File: database/migrations/20221201070519-user-migrations.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        allowNull: false,
+      },
+      phone_number: {
+        type: Sequelize.STRING,
+        allowNull: false,
```
Comment (kelue) [inline database/migrations/20221201070519-user-migrations.js:28]: same with phone number entries should be unique

File: database/migrations/20221201070519-user-migrations.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        defaultValue: false,
+        allowNull: false,
+      },
+      is_active: {
+        type: Sequelize.BOOLEAN,
+        defaultValue: false,
+        allowNull: false,
+      },
+      role: {
+        type: Sequelize.ENUM('Admin', 'User'),
```
Comment (kelue) [inline database/migrations/20221201070519-user-migrations.js:41]: role should have default value of `user`

**Review by kelue — DISMISSED** [review]

File: database/migrations/20221201070519-user-migrations.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+/* eslint-disable no-unused-vars */
+/** @type {import('sequelize-cli').Migration} */
```
Comment (Endowmissy) [inline database/migrations/20221201070519-user-migrations.js]: @jonyoung123 what's this for

File: database/migrations/20221201070519-user-migrations.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+module.exports = {
+  async up(queryInterface, Sequelize) {
```
Comment (Endowmissy) [inline database/migrations/20221201070519-user-migrations.js]: @jonyoung123 modify these to use arrow function

File: database/migrations/20221201070519-user-migrations.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    await queryInterface.createTable('users', {
+      id: {
+        type: Sequelize.UUID,
+        defaultValue: Sequelize.UUIDV4,
+        primaryKey: true,
+        // autoIncrement: true,
```
Comment (Endowmissy) [inline database/migrations/20221201070519-user-migrations.js]: remove this

File: database/migrations/20221201070519-user-migrations.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+      role: {
+        type: Sequelize.ENUM('Admin', 'User'),
+        allowNull: false,
+      },
+    });
+  },
+
+  async down(queryInterface, Sequelize) {
+    await queryInterface.dropTable('users');
+    /**
```
Comment (Endowmissy) [inline database/migrations/20221201070519-user-migrations.js]: remove this

**Comment by kelue** [comment]
Add proper description to pull request describing what the pull request does and the steps to testing

File: database/migrations/20221201070519-user-migrations.js (hunk 5 above)
Comment (jonyoung123) [reply database/migrations/20221201070519-user-migrations.js]: automatically generated from sequelize cli commands

File: database/migrations/20221201070519-user-migrations.js (hunk 1 above)
Comment (jonyoung123) [reply database/migrations/20221201070519-user-migrations.js]: Noted.  it's resolved now, thanks.

File: database/migrations/20221201070519-user-migrations.js (hunk 7 above)
Comment (jonyoung123) [reply database/migrations/20221201070519-user-migrations.js]: I'm to remove the "id" field completely?

File: database/migrations/20221201070519-user-migrations.js (hunk 8 above)
Comment (jonyoung123) [reply database/migrations/20221201070519-user-migrations.js]: it's there in case the table is needed to be dropped from the database. used for testing

**Review by Endowmissy — APPROVED** [review]

**Review by bright258 — APPROVED** [review]

**Pull request merged**